plugins/CircuitViewer/io/CircuitLoader.cpp

```diff
@@ -261,23 +299,42 @@ class Impl
                                   {"mesh-folder", _properties.meshFolder}};
         auto model = _scene.createModel();
 
-        // Open Circuit and select GIDs according to specified target
-        const brion::BlueConfig bc(source);
-        const brain::Circuit circuit(bc);
+        // Open circuit and simulation if possible
+        brion::URI uri(source);
+        std::unique_ptr<brain::Simulation> simulation;
+        try
+        {
+            simulation = std::make_unique<brain::Simulation>(uri);
+        }
+        catch (std::runtime_error& error)
+        {
+            BRAYNS_INFO << "Cannot open " + source + " as simulation config: "
+                        << error.what() << std::endl;
+        }
+        const auto circuit =
+            simulation ? simulation->openCircuit() : brain::Circuit(uri);
 
         brain::GIDSet allGids;
         GIDOffsets targetGIDOffsets;
         targetGIDOffsets.push_back(0);
 
-        strings localTargets;
-        if (_properties.targetList.empty())
-            localTargets.push_back(bc.getCircuitTarget());
-        else
-            localTargets = _properties.targetList;
+        const strings localTargets = _properties.targetList.empty()
+                                         ? strings{{""}}
+                                         : _properties.targetList;
+
+        auto resolver = [&](const std::string& name) {
+            const float fraction = _properties.density / 100.0f;
+            if (simulation)
+                return simulation->getGIDs(name, fraction,
+                                           _properties.randomSeed);
+            return circuit.getRandomGIDs(fraction, name,
```

Comment:
nice that the signature is not the same for those two getID functions :stuck_out_tongue:

Reply:
I was referring to `name, fraction` vs `fraction, name` argument swap. Asking the simulation vs the circuit is also a valid point.